config/hotreloadable.go

```diff
@@ -1,14 +1,74 @@
 package config
 
-import "time"
+import (
+	"strings"
+	"sync"
+	"time"
+)
 
 // RegisterIntConfigVariable registers int config variable
-func RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, keys ...string) {
-	Default.RegisterIntConfigVariable(defaultValue, ptr, isHotReloadable, valueScale, keys...)
+// WARNING: keys are being looked up in requested order and the value of the first found key is returned,
+// e.g. asking for the same keys but in a different order can result in a different value to be returned
+//
+// Deprecated: use RegisterIntVar or RegisterReloadableIntVar instead
+func RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, orderedKeys ...string) {
+	Default.RegisterIntConfigVariable(defaultValue, ptr, isHotReloadable, valueScale, orderedKeys...)
+}
+
+// RegisterIntVar registers a not hot-reloadable int config variable
+// WARNING: keys are being looked up in requested order and the value of the first found key is returned,
+// e.g. asking for the same keys but in a different order can result in a different value to be returned
+func RegisterIntVar(defaultValue, valueScale int, orderedKeys ...string) int {
+	return Default.RegisterIntVar(defaultValue, valueScale, orderedKeys...)
+}
+
+// RegisterReloadableIntVar registers a hot-reloadable int config variable
+// WARNING: keys are being looked up in requested order and the value of the first found key is returned,
+// e.g. asking for the same keys but in a different order can result in a different value to be returned
+func RegisterReloadableIntVar(defaultValue, valueScale int, orderedKeys ...string) *Reloadable[int] {
+	return Default.RegisterReloadableIntVar(defaultValue, valueScale, orderedKeys...)
 }
 
 // RegisterIntConfigVariable registers int config variable
-func (c *Config) RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, keys ...string) {
+// WARNING: keys are being looked up in requested order and the value of the first found key is returned,
+// e.g. asking for the same keys but in a different order can result in a different value to be returned
+//
+// Deprecated: use RegisterIntVar or RegisterReloadableIntVar instead
+func (c *Config) RegisterIntConfigVariable(
+	defaultValue int, ptr *int, isHotReloadable bool, valueScale int, orderedKeys ...string,
+) {
+	c.registerIntVar(defaultValue, ptr, isHotReloadable, valueScale, func(v int) {
+		*ptr = v
+	}, orderedKeys...)
+}
+
+// RegisterIntVar registers a not hot-reloadable int config variable
+// WARNING: keys are being looked up in requested order and the value of the first found key is returned,
+// e.g. asking for the same keys but in a different order can result in a different value to be returned
+func (c *Config) RegisterIntVar(defaultValue, valueScale int, orderedKeys ...string) int {
```

Comment:
**Question:** for the new api, should we use `GetIntVar` & `GetReloadableIntVar` instead of `RegisterIntVar` & `RegisterReloadableIntVar`

Reply:
Good question, I think I prefer the `GetXXX` version, I think after the latest changes it makes more sense :+1: